### PR TITLE
Add truffle migrations for Goerli and Mumbai

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
Adds migration files with configs for Goerli and Mumbai testnets (also makes inital configurations - registers resources, approvals,...)

## Related Issue Or Context
Closes: #15 

## How Has This Been Tested? Testing details.
Manual tests

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

**Review by mpetrun5 — COMMENTED** [review]
Would be nice to have this deployed via Github CI on release tag.

**Review by P1sar — CHANGES_REQUESTED** [review]

File: migrations/2_deploy_goerli.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+const Ethers = require("ethers");
+
+const networksConfig = require("./networks_config");
+const Helpers = require('../test/helpers');
+
+const AccessControlSegregatorContract = artifacts.require("AccessControlSegregator");
+const PausableContract = artifacts.require("Pausable");
+const BridgeContract = artifacts.require("Bridge");
+const CentrifugeAssetContract = artifacts.require("CentrifugeAsset");
```
[the remark this answers is not in the document]
Comment (mpetrun5) [reply migrations/2_deploy_goerli.js:14]: Was thinking of creating a docker image built on chainbridge-solidity with preloaded configuration to use on sygma-relayer to remove the horrible e2e testing waiting time so will probably need Centrifuge assets for testing generic handler. 1155 is not supported so not really needed right now.

**Comment by lastperson** [comment]
Why the deployment scripts differ on a network basis? Shouldn't only configuration change and script be the same? At least that is how truffle migrations are supposed to be used.

File: migrations/networks_config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+    "domainID": 5,
+    "erc721Name": "ERC721Test",
+    "erc721Symbol": "ERC721TST",
+    "erc721URI": "metadata.test",
+    "erc20Name": "ERC20Test",
+    "erc20Symbol": "ERC20TST",
```
Comment (P1sar) [inline migrations/networks_config.js:7]: we have 2 erc20 tokens for devnet for example

File: .env.sample [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+MUMBAI_PROVIDER_URL="provider url"
+MUMBAI_API_KEY="provider api key"
+# configure token mint/approve amount and fee handlers
+TOKEN_AMOUNT="93274"
+FEE_ORACLE_ADDRESS="0x534aB5328d0447d6B13c5837D4F3C95ee0F108A2"
+# fee handler with oracle
+FHWO_GAS_USED="4444"
+FHWO_FEE_PERCENTAGE=666
+# basic fee handler, fee is in ETH
+BFH_FEE=0.3
```
Comment (mpetrun5) [inline .env.sample:14]: Can we please set this to something more reasonable as 0.001?  We don't have that much test tokens :smile: 
Also, check other values so they make sense. (Token amount 93274?)

File: migrations/2_deploy_contracts.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        "BasicFeeHandler Address": basicFeeHandlerInstance.address,
+        "FeeHandlerWithOracle Address": feeHandlerWithOracleInstance.address,
+        "ERC20 resourceID": currentNetworkConfig.erc20ResourceID,
+        "Generic resourceID": currentNetworkConfig.genericResourceID,
+        "ERC721 resourceID": currentNetworkConfig.erc721ResourceID,
+        "ERCLockRelease20 resourceID": currentNetworkConfig.erc20LockReleaseResourceID
+    });
+
+    /* setup contracts */
+    const tokenAmount = process.env.TOKEN_AMOUN || 10000;
```
Comment (mpetrun5) [inline migrations/2_deploy_contracts.js:89]:
```suggestion
    const tokenAmount = process.env.TOKEN_AMOUNT || 10000;
```

File: .env.sample (hunk 3 above)
Comment (mpetrun5) [reply .env.sample:14]: Or are these values not used at all and are just an example. Forgot how env sample works.

File: migrations/2_deploy_contracts.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    const feeOracleAddress = process.env.FEE_ORACLE_ADDRESS || "0x70B7D7448982b15295150575541D1d3b862f7FE9";
+    const feeHandlerWithOracleGasUsed = process.env.FHWO_GAS_USED || 100000;
+    const feeHandlerWithOracleFeePercentage = process.env.FHWO_FEE_PERCENTAGE || 500; // 5%
+    const basicFeeHandlerFee = Ethers.utils.parseEther(process.env.BFH_FEE || "0.1");
```
Comment (mpetrun5) [inline migrations/2_deploy_contracts.js:93]: Set to something smaller

**Review by mpetrun5 — CHANGES_REQUESTED** [review]

**Review by mpetrun5 — APPROVED** [review]

**Review by MakMuftic — APPROVED** [review]

File: migrations/networks_config.json [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+  },
+  "mumbai": {
+    "domainID": "3",
+    "erc721Name": "ERC721Test",
+    "erc721Symbol": "ERC721TST",
+    "erc721URI": "metadata.test",
+    "erc20Name": "ERC20Test",
+    "erc20Symbol": "ERC20TST",
+    "erc20LRName": "ERC20Test",
+    "erc20LRSymbol": "ERC20TST",
```
Comment (P1sar) [inline migrations/networks_config.json:40]: what  *LR* means? Why symbols are the same?

Comment (P1sar) [reply migrations/networks_config.json:40]: ah ok lock release, but lets make symbols different anyway

**Review by P1sar — CHANGES_REQUESTED** [review]
looks fine except token symbols

File: migrations/networks_config.json (hunk 6 above)
Comment (nmlinaric) [reply migrations/networks_config.json:40]: Yeah, missed it :sweat_smile:

**Review by P1sar — APPROVED** [review]

**Pull request merged**